pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java

```diff
@@ -133,13 +136,23 @@ public SuccessResponse delete(
       @ApiResponse(code = 500, message = "Internal server error")
   })
   public SuccessResponse putData(
-      @ApiParam(value = "Zookeeper Path, must start with /", required = true, defaultValue = "/") @QueryParam("path")
+      @ApiParam(value = "Zookeeper Path, must start with /", defaultValue = "/") @QueryParam("path")
       @DefaultValue("") String path,
-      @ApiParam(value = "Content", required = true) @QueryParam("data") @DefaultValue("") String content,
-      @ApiParam(value = "expectedVersion", required = true, defaultValue = "-1") @QueryParam("expectedVersion")
+      @ApiParam(value = "Content") @QueryParam("data") @DefaultValue("") String content,
+      @ApiParam(value = "expectedVersion", defaultValue = "-1") @QueryParam("expectedVersion")
       @DefaultValue("-1") String expectedVersion,
-      @ApiParam(value = "accessOption", required = true, defaultValue = "1") @QueryParam("accessOption")
-      @DefaultValue("1") String accessOption) {
+      @ApiParam(value = "accessOption", defaultValue = "1") @QueryParam("accessOption")
+      @DefaultValue("1") String accessOption,
+      @Nullable Payload payload
+  ) {
+
+    if (payload != null) {
+      path = payload.getPath();
```

Comment:
We should only override the value when it is provided in the payload

Reply:
@Jackie-Jiang updated the requested changes